cmd/embedded-cluster/reset.go

```diff
@@ -147,151 +216,134 @@ func (h *hostInfo) stopAndResetK0s() error {
 	if err != nil {
 		return fmt.Errorf("could not reset k0s: %w, %s", err, string(out))
 	}
-	fmt.Println("Node has been reset, please reboot to ensure transient configuration is also reset")
+	fmt.Println("Node has been reset. Please reboot to ensure transient configuration is also reset.")
 	return nil
 }
 
 // newHostInfo returns a populated hostInfo struct
-func newHostInfo(ctx context.Context) (hostInfo, error) {
+func newHostInfo(c *cli.Context) (hostInfo, error) {
 	currentHost := hostInfo{}
 	// populate hostname
 	err := currentHost.getHostName()
 	if err != nil {
 		return currentHost, err
 	}
-	// set up kube client
-	err = currentHost.configureKubernetesClient()
+	// get k0s status json
+	out, err := exec.Command(k0s, "status", "-o", "json").Output()
 	if err != nil {
 		return currentHost, err
 	}
-	// fetch node object
-	err = currentHost.getNodeObject(ctx)
+	err = json.Unmarshal(out, &currentHost.Status)
 	if err != nil {
 		return currentHost, err
 	}
+	currentHost.RoleName = currentHost.Status.Role
+	// set up kube client
+	currentHost.configureKubernetesClient()
+	// fetch node object
+	currentHost.getNodeObject(c.Context)
 	// control plane only stff
-	if currentHost.isControlPlane() {
+	if currentHost.Status.Role == "controller" {
 		// fetch controlNode
-		err := currentHost.getControlNodeObject(ctx)
-		if err != nil {
-			return currentHost, err
-		}
+		currentHost.getControlNodeObject(c.Context)
+	}
+	// try and get custom role name from the node labels
+	labels := currentHost.Node.GetLabels()
+	if value, ok := labels["kots.io/embedded-cluster-role-0"]; ok {
+		currentHost.RoleName = value
 	}
 	return currentHost, nil
 }
 
-func checkErrPrompt(err error) bool {
+func checkErrPrompt(c *cli.Context, err error) bool {
 	if err == nil {
 		return true
 	}
-	fmt.Println("-----")
 	fmt.Println(err)
-	fmt.Println("-----")
-	fmt.Println("An error has occured while trying to reset this node.")
-	fmt.Println("Continuing may leave the cluster in an unexpected state")
+	if c.Bool("force") || c.Bool("no-prompt") {
+		return true
+	}
+	fmt.Println("An error occurred while trying to reset this node.")
```

Comment:
Sorry if I'm totally missing it, but I thought `no-prompt` was going to bail if an error occurred, rather than just continuing through it. It would just skip confirmation type prompts. (I could have misunderstood too.)

Reply:
No you're correct, that's the intended result. 

looking into this

Reply:
Should be fixed